doc/languages-frameworks/javascript.section.md

```diff
@@ -909,6 +909,84 @@ stdenv.mkDerivation (finalAttrs: {
 })
 ```
 
+### fetchDenoDeps {#javascript-fetchDenoDeps}
+
+`fetchDenoDeps` allows you to create a derivation containing all dependencies needed to run a [Deno](https://deno.com/) package.
+
+### Usage {#javascript-fetchDenoDeps-usage}
+
+1. Define the derivation for the dependencies like this:
+
+    ```nix
+    { fetchDenoDeps }:
+    {
+      my-deps = fetchDenoDeps {
+        name = "<name>";
+        denoLock = ./path/to/lockfile;
+        hash = "<hash>";
+      };
+    }
+    ```
+
+
+2. Use the deps build like this:
+
+    ```nix
+    { stdenvNoCC, my-deps }:
+    {
+      my-deno-package = stdenvNoCC.mkDerivation {
+        name = "<name>";
+        src = ./path/to/project/src;
+        buildPhase = ''
+          # copy the deps to the required location
+          cp -r --no-preserve=mode ${my-deps.denoDeps}/* ./
```

Comment:
Should `shopt -s dotglob` be used to also copy the `.deno` folder?

without it I got:
```
       > error: Failed caching npm package 'optics-ts@2.4.1'
       >     0: npm package not found in cache: "optics-ts", --cached-only is specified.
```